### PR TITLE
Adding input path to runtime. Changing default split mesh file prefix.

### PR DESCRIPTION
**Pull request by wdhawkins**
This PR adds the input path to the openSn runtime as a `std::filesystem::path` object. This allows us to use the input path and filename stem throughout the code. In this case, I'm using the input filename stem as the default prefix for split mesh files. This prevents a whole host of issues when using the same filename prefix across multiple runs. IMHO, we should be using the input file stem for all files generated by the code instead of generic names that lack provenance.

This should fix the race condition encountered with the split file regression tests, and I have re-enabled the offending test.

There may already be a Lua method for grabbing the input file path from the command-line.  I'm not very familiar with Lua or the Lua interface yet. If such a method exists, I'm happy to use it in place of the changes I've made here.

### CODE REVIEW

**Review by ragusa — COMMENTED** [review]
Good for me. But I must admit it looks a tiny bit obfuscated. Before, the meaning was crystal clear with the variable input_file_name but if this is what you have to do ... I looked a bit into lua for this, I found this:
https://stackoverflow.com/questions/6380820/get-containing-path-of-lua-file
which uses standard modules in lua. What do you guys think? @andrsd @zhardy-lanl

**Comment by wdhawkins** [comment]
> #141 was also patching `framework/mesh/mesh_generator/split_file_mesh_generator.cc` which changed the `split_file_prefix` parameter in `SplitFileMeshGenerator` and did some other changes related to parallel execution. This bit seems to be missing here. Is this PR complete? Just checking...

Yes, this PR is complete. I stripped out the performance piece because I discovered a failure mode at high core counts. I'll create a separate PR for it at a later date. It wasn't really part of the bug fix anyway and should have a separate PR.

**Comment by wdhawkins** [comment]
> Good for me. But I must admit it looks a tiny bit obfuscated. Before, the meaning was crystal clear with the variable input_file_name but if this is what you have to do ... I looked a bit into lua for this, I found this: https://stackoverflow.com/questions/6380820/get-containing-path-of-lua-file which uses standard modules in lua. What do you guys think? @andrsd @zhardy-lanl

I can go back to input_file_name. To me, that's not as descriptive, but I'm ok with it... especially if you think it's clearer and easier to understand.

**Comment by zhardy-lanl** [comment]
Am I understanding this correctly when I make the comparison that using `std::filesystem` essentially just objectifies a path like `Pathlib` in Python does? This doesn't seem to change anything on the input side and seems to offer more benefits than just using a string for the input path.  If this is true, I like this change

**Comment by zhardy-lanl** [comment]
> Good for me. But I must admit it looks a tiny bit obfuscated. Before, the meaning was crystal clear with the variable input_file_name but if this is what you have to do ... I looked a bit into lua for this, I found this: https://stackoverflow.com/questions/6380820/get-containing-path-of-lua-file which uses standard modules in lua. What do you guys think? @andrsd @zhardy-lanl

It doesn't look like this changes anything in Lua. See my last comment. It seems like we are just replacing the C++ `input_file_name` with a path-like C++ object in `std::filesystem` that has convenient routines available. I don't think this looks obfuscated, personally. We are just representing a path differently.

**Comment by wdhawkins** [comment]
> Am I understanding this correctly when I make the comparison that using `std::filesystem` essentially just objectifies a path like `Pathlib` (or kind of `os`) in Python does? This doesn't seem to change anything on the input side and seems to offer more benefits than just using a string for the input path. If this is true, I like this change

Yes, that is correct. Treating input path as a std::filesystem path object provides us with a number of benefits: dissecting the path into individual pieces, checking if the path exists, etc. Much more useful than just a string.

**Review by zhardy-lanl — APPROVED** [review]

**Review by andrsd — DISMISSED** [review]
This is good. Just the title is misleading. This PR is not "Changing default split mesh file prefix".

**Comment by andrsd** [comment]
I approved, but please change the title before merging. It is good for future us searching through PRs.

**Comment by wdhawkins** [comment]
> I approved, but please change the title before merging. It is good for future us searching through PRs.

That was my fault. When I removed the changes to the split mesh file generation algorithm, I also removed the change that update the file prefix. I've add that change back into the patch. It's also necessary for re-enabling the regression test.

**Review by andrsd — CHANGES_REQUESTED** [review]
Merge the formatting patch with `ab531c3`. The usual business:

```
git rebase --interactive main
- mark the formatting patch as fixup
- save and exit
git push -f
```

**Comment by wdhawkins** [comment]
Just one additional note. I know that most of the methods that create output files have the option to take a file prefix (just like the split mesh routine). IMHO, we should default to using the input file stem as the prefix. The user can certainly change the prefix if necessary, but ,this way, there is a clear association between output files and input files without any manual intervention by the user.

**Review by andrsd — APPROVED** [review]
:+1:

**Review by ragusa — APPROVED** [review]

File: framework/mesh/mesh_generator/split_file_mesh_generator.cc
Comment (zhardy-lanl) [inline framework/mesh/mesh_generator/split_file_mesh_generator.cc:43]: I would recommend changing this to just `file_prefix` since this is always within the block for creating s split file mesh.

**Review by zhardy-lanl — APPROVED** [review]
Please see my recommended input parameter name change. I won't hold up the PR for the change, but please consider it before merging.

**Pull request merged**
